CMakeLists.txt

```diff
@@ -132,6 +132,38 @@ if (CMAKE_BUILD_TYPE MATCHES "Coverage")
     COMMAND ${CMAKE_SOURCE_DIR}/scripts/record-coverage.sh ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR})
 endif()
 
+# Enable asserts for C/C++ code if requested
+#  from the LLVM project, https://opensource.apple.com/source/llvmCore/llvmCore-2358.3/CMakeLists.txt.auto.html
+if(CMAKE_BUILD_TYPE STREQUAL "Debug" )
+    option(PN_ENABLE_ASSERTIONS "Enable assertions" ON)
+else()
+    option(PN_ENABLE_ASSERTIONS "Enable assertions" OFF)
+endif()
+if(PN_ENABLE_ASSERTIONS)
```

Comment:
Please call this ``ENABLE_ASSERTIONS`` - we don't use a PN_ prefix as everything in the build is Proton!

Reply:
Ack, will rename.

It has the same justification as the `pn_` prefix in the C library. Somebody might include the CMake build in their own project and then there would be value in being able to configure options for each project separately. I admit that here is a poor place where to start introducing the convention. Especially since I dont intend to apply it uniformly everywhere.